jadx-core/src/main/java/jadx/core/dex/nodes/utils/ClassUtils.java

```diff
@@ -14,44 +12,7 @@ public ClassUtils(RootNode rootNode) {
 	}
 
 	public boolean isAccessible(ClassNode cls, ClassNode callerCls) {
-		if (cls.equals(callerCls)) {
-			return true;
-		}
-
-		final AccessInfo accessFlags = cls.getAccessFlags();
-		if (accessFlags.isPublic()) {
-			return true;
-		}
-
-		if (accessFlags.isProtected()) {
-			return isProtectedAccessible(cls, callerCls);
-		}
-
-		if (accessFlags.isPackagePrivate()) {
-			return isPackagePrivateAccessible(cls, callerCls);
-		}
-
-		if (accessFlags.isPrivate()) {
-			return isPrivateAccessible(cls, callerCls);
-		}
-
-		throw new JadxRuntimeException(accessFlags + " is not supported");
-	}
-
-	private boolean isProtectedAccessible(ClassNode cls, ClassNode callerCls) {
-		return isPackagePrivateAccessible(cls, callerCls) || isSuperType(cls, callerCls);
-	}
-
-	private boolean isPackagePrivateAccessible(ClassNode cls, ClassNode callerCls) {
-		return cls.getPackageNode().equals(callerCls.getPackageNode());
-	}
-
-	private boolean isPrivateAccessible(ClassNode cls, ClassNode callerCls) {
-		return cls.getTopParentClass().equals(callerCls.getTopParentClass());
-	}
-
-	private boolean isSuperType(ClassNode cls, ClassNode superCls) {
-		return root.getClsp().getSuperTypes(cls.getRawName()).stream()
-				.anyMatch(x -> x.equals(superCls.getRawName()));
+		CodeNodeUtils codeNodeUtils = new CodeNodeUtils(root);
```

Comment:
I don't see a reason to create this instance every time. Please store it in final field (same for method utils)
Also, maybe name it something like `AccessChecks` or `AccessCheckUtils` it is big enough and more specific name is better.

Reply:
Done